pygatt/bled112_backend.py

```diff
@@ -0,0 +1,1977 @@
+                self._logger.debug("char 0x%s --> %s", char_uuid_str,
+                                   hex(char_obj.handle))
+                for desc_uuid_str, desc_handle in (
+                        char_obj.descriptors.iteritems()):
+                    self._logger.debug("desc 0x%s --> %s", desc_uuid_str,
+                                       hex(desc_handle))
+
+        # Return the handle if it exists
+        char = None
+        char_uuid_str = hexlify(characteristic_uuid)
+        if not (char_uuid_str in self._characteristics):
+            self._logger.warn("No such characterisitic")
+            self._loglock.release()
+            self._main_thread_cond.release()
+            return
+        char = self._characteristics[char_uuid_str]
+        if descriptor_uuid is None:
+            self._loglock.release()
+            self._main_thread_cond.release()
+            return char.handle
+        desc_uuid_str = hexlify(descriptor_uuid)
+        if not (desc_uuid_str in char.descriptors):
+            self._logger.warn("No such descriptor")
+            self._loglock.release()
+            self._main_thread_cond.release()
+            return
+        desc_handle = char.descriptors[desc_uuid_str]
+        self._loglock.release()
+        self._main_thread_cond.release()
+        return desc_handle
+
+    def get_notifications(self):
+        """
+        Get self._notifications in a thread-safe way.
+
+        Returns the self._notifications dictionary.
+        """
+        # get locks
+        self._main_thread_cond.acquire()
+        self._loglock.acquire()
+
+        # Log
+        self._logger.info("get_notifications")
+
+        notifications = self._notifications
+
+        # Drop locks
+        self._loglock.release()
+        self._main_thread_cond.release()
+
+        return notifications
+
+    def get_rssi(self):
+        """
+        Get the receiver signal strength indicator (RSSI) value from the device.
+
+        This requires that a connection is already established with the device.
+
+        Returns the RSSI as in integer in dBm.
+        """
+        # Get locks
+        self._main_thread_cond.acquire()
+        self._loglock.acquire()
+
+        # Make sure there is a connection
+        if not self._connected:
+            self._logger.warn("Not connected")
+            self._loglock.release()
+            self._main_thread_cond.release()
+            return
+
+        # Get RSSI value
+        self._logger.info("get_rssi")
+        cmd = self._lib.ble_cmd_connection_get_rssi(self._connection_handle)
+        self._lib.send_command(self._ser, cmd)
+
+        # Wait for response
+        self._loglock.release()  # don't hold loglock while waiting
+        while not self._response_received:
+            self._main_thread_cond.wait()
+        self._response_received = False  # reset the flag
+        rssi_value = self._response_return
+
+        # Drop locks
```

Comment:
Nope because the lock isn't held at that point. Just need to change the comment.